eta/core/data.py

```diff
@@ -827,9 +827,31 @@ def cull(self, field, keep_values=None, remove_values=None):
                 "Either keep_values or remove_values must be provided")
 
         # Cull records
-        inds = []
+        inds = set()
```

Comment:
It's not clear to me why you changes inds from a list to a set.  The `self.keep_inds` function from `Container` expects a list.  (Actually, that function in line 440 of `core/serial.py` now converts the input to a set.  Perhaps the interface to keep_inds be redefined to require a set in its contract and this conversion should only happen once.  Is a `set` more efficient for the list comprehension in `keep_inds`?)

Reply:
`set` is more efficient both here and in `keep_inds` because sets are implemented as hash maps with `O(1)` lookups.

Strictly speaking, there are many places in our code where we document that the arg should be a `list` when we all we do is iterate over it and thus it could be any `iterable`. Not requesting any action on this front, just commenting.

Reply:
Agreed.  I propose changing the doc string for `inds` to specify an `iterable` rather than `list`, and will make that change unless there are any objections.